src/js/player.js

```diff
@@ -1241,7 +1241,7 @@ class Player extends Component {
     }
 
     // update `currentSource` cache always
-    this.cache_.source = {src, type};
+    this.cache_.source = mergeOptions({}, srcObj, {src, type});
```

Comment:
mergeOptions ({}, srcObj, { src, type} nie wiem czy to bezpieczne i teraz mi się że zaraz po dziejesz sytuacji w której znajduje się w której nie wiem czy to bezpieczne